### PR TITLE
Update folder structure

### LINKED ISSUE #2: Change folder structure
Move `./src/system` to `./src` and `./src` to `./lib`.
In short:
- `./src` is for application loader and server (transports)
- `./lib` is for common code

### PR DESCRIPTION
**Pull request by mokuteki225**
Closes #2

### CODE REVIEW

**Comment by tshemsedinov** [comment]
What we need moving files, also change `require` paths and run `npm t` before commit @mokuteki225

**Comment by tshemsedinov** [comment]
For example, here is a PR for pure node.js repo: https://github.com/metatech-university/NodeJS-Pure/pull/1/files

File: src/logger.js [hunk 1]
```diff
@@ -17,12 +17,12 @@ class Logger {
     this.logger.info(msg);
   }
 
-  debug(...args) {
+  debug(data, ...args) {
     const msg = args.length <= 1 ? args[0] : args;
     this.logger.debug(data, msg);
   }
```
Comment (tshemsedinov) [inline src/logger.js:23]: `data` is not used below in this method

File: src/logger.js [hunk 2]
```diff
@@ -17,12 +17,12 @@ class Logger {
...
 
-  error(...args) {
```
Comment (tshemsedinov) [inline src/logger.js:25]: ditto

**Review by tshemsedinov — CHANGES_REQUESTED** [review]

File: src/logger.js (hunk 1 above)
Comment (mokuteki225) [reply src/logger.js:23]: Take a look at line 22

File: src/logger.js [hunk 3]
```diff
@@ -17,12 +17,12 @@ class Logger {
     this.logger.info(msg);
   }
 
-  debug(...args) {
+  debug(data, ...args) {
     const msg = args.length <= 1 ? args[0] : args;
     this.logger.debug(data, msg);
```
Comment (mokuteki225) [inline src/logger.js:22]: This line was a part of the codebase before my PR, `npm t` fails with the following message 
**_"22:23  error  'data' is not defined  no-undef"_**
So we have to either just call `this.logger.debug(msg);`, or get the `data` as an argument

Comment (tshemsedinov) [reply src/logger.js:22]: Yes thanks! We need to notify @nechaido to avoid this bug in other implementations/versions of this codebase

Comment (nechaido) [reply src/logger.js:22]: Consider him notified.

**Pull request merged**